### PR TITLE
Use the appsv1 apigroup when building the reflector for replicasets.

### LINKED ISSUE #815: ReplicaSet unexpected type in v1.7.0-rc.0
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

Logs showed:

```
E0710 17:49:42.568957       1 reflector.go:342] k8s.io/kube-state-metrics/internal/store/builder.go:295: expected type *v1beta1.ReplicaSet, but watch event object had type *v1.ReplicaSet
```

**What you expected to happen**:

Only non-error logs.

**How to reproduce it (as minimally and precisely as possible)**:

Always with v1.7.0-rc.0

**Anything else we need to know?**:

n/a

**Environment**:
- Kubernetes version (use `kubectl version`): v1.15.0 minikube
- Kube-state-metrics image version: v1.7.0-rc.0

@tariq1890 @LiliC @mxinden @andyxning

### PR DESCRIPTION
**Pull request by tariq1890**
This fixes #815

### CODE REVIEW

**Comment by brancz** [comment]
Can you add a changelog entry and bump to rc.1 already?

**Comment by tariq1890** [comment]
Oh yes, will do!

**Comment by tariq1890** [comment]
Done!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
-## v1.7.0-rc.0 / 2019-07-10
+## v1.7.0-rc.1 / 2019-07-10
 
+* [BUGFIX] Use the correct apigroup when building the reflector for replicasets
```
Comment (brancz) [inline CHANGELOG.md]: Make this a new one, and keep rc.0

Comment (tariq1890) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
-## v1.7.0-rc.0 / 2019-07-10
+## v1.7.0-rc.1 / 2019-07-10
+
+* [BUGFIX] Use the appsv1 apigroup when building the reflector for replicasets. #816
+
+## v1.7.0-rc.1 / 2019-07-10
```
Comment (brancz) [inline CHANGELOG.md]: this needs to be reverted to rc.0

Comment (tariq1890) [reply CHANGELOG.md]: Thanks for catching that.

**Comment by brancz** [comment]
cc @serathius @piosz @DirectXMan12 please stand by to publish the gcr.io image :)

**Comment by tariq1890** [comment]
@brancz Can we remove the hold?

**Comment by LiliC** [comment]
@tariq1890 this was not merged into master.

**Comment by LiliC** [comment]
@tariq1890 @brancz I believe it's been 7 days since the release candidate and no new issues were reported so we can do a final release? https://github.com/kubernetes/kube-state-metrics/releases/tag/v1.7.0-rc.1

**Comment by tariq1890** [comment]
No objections from me. Feel free to create a PR to update the release 1.7 branch if you wish. Once @brancz approves and pushes the stable image to quay and gcr, we can publish the stable release.

**Pull request merged**